### DISCUSSION TITLE
Visual Studio 16.4 vs 16.5

### DISCUSSION THREAD

**Discussion by steven-johnson** [community/community#26194] [Actions]
<p>The current Windows VMs appear to have VS 16.4 installed; unfortunately, this version is known to be incompatible with LLVM; attempting to build LLVM with it fails with <code>Host Visual Studio version 16.4 is known to miscompile part of LLVM, please use clang-cl or upgrade to 16.5 or above (use -DLLVM_TEMPORARILY_ALLOW_OLD_TOOLCHAIN=ON to ignore)</code>.</p>
<p>While the workaround above is obvious, it’s also a problem (the error is there for a reason). Is there a plan/timetable to upgrade the VMs to VS 16.5?</p>
<p>(For more info, see: <a href="https://developercommunity.visualstudio.com/content/problem/845933/miscompile-boolean-condition-deduced-to-be-always.html" rel="nofollow noopener">https://developercommunity.visualstudio.com/content/problem/845933/miscompile-boolean-condition-deduced-to-be-always.html</a>)</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi steven-johnson, </p>
<p>I checked visual studio 16.5, it is still in preview version. <a href="https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes-preview" rel="nofollow noopener">https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes-preview </a></p>
<p>When a stable version of VS16.5 published, the Hosted runner on windows vm will update to use it . But I am not sure the timetable for it. As a temporary work around , you could consider using <a href="https://help.github.com/en/actions/hosting-your-own-runners/adding-self-hosted-runners" rel="nofollow noopener">self-hosted runner</a> with suitable VS version . </p>

**Comment by alexreinking** [comment]
<p>For some context, this is a discussion on Visual Studio’s product forums about the bug: <a href="https://developercommunity.visualstudio.com/content/problem/845933/miscompile-boolean-condition-deduced-to-be-always.html" rel="nofollow noopener">https://developercommunity.visualstudio.com/content/problem/845933/miscompile-boolean-condition-deduced-to-be-always.html</a></p>
